src/corehost/common/pal.windows.cpp

```diff
@@ -134,48 +134,50 @@ void pal::unload_library(dll_t library)
     // No-op. On windows, we pin the library, so it can't be unloaded.
 }
 
-bool pal::get_default_breadcrumb_store(string_t* recv)
-{
-    recv->clear();
-
-    pal::char_t* prog_dat;
-    HRESULT hr = ::SHGetKnownFolderPath(FOLDERID_ProgramData, 0,  NULL, &prog_dat);
-    if (hr != S_OK)
-    {
-        trace::verbose(_X("Failed to read default breadcrumb store 0x%X"), hr);
-        return false;
-    }
-    recv->assign(prog_dat);
-    append_path(recv, _X("Microsoft"));
-    append_path(recv, _X("NetFramework"));
-    append_path(recv, _X("BreadcrumbStore"));
-    return true;
-}
-
 static
-bool get_program_files_by_id(KNOWNFOLDERID kfid, pal::string_t* recv)
+bool get_file_path_from_env(const pal::char_t* kfid, pal::string_t* recv)
 {
     recv->clear();
-    pal::char_t* prog_files;
-    HRESULT hr = ::SHGetKnownFolderPath(kfid, 0,  NULL, &prog_files);
-    if (hr != S_OK)
+    pal::string_t file_path;
+    if (!(pal::getenv(kfid, &file_path) && pal::realpath(&file_path)))
     {
-        trace::verbose(_X("Failed to obtain Program Files directory, HRESULT: 0x%X"), hr);
+        // We should have the path in file_path.
+        trace::verbose(_X("Failed to obtain [%s] directory,[%s]"),kfid, file_path.c_str());
         return false;
     }
-    recv->assign(prog_files);
+
+    recv->assign(file_path);
     return true;
 }
 
 static
 bool get_wow_mode_program_files(pal::string_t* recv)
 {
 #if defined(_TARGET_AMD64_)
-    KNOWNFOLDERID kfid = FOLDERID_ProgramFilesX86;
+    pal::char_t* kfid = _X("ProgramFiles(x86)");
```

Comment:
Since this is no longer KNOWNFOLDERID, can you please update the variable name to reflect it?